src/pollyCLI/pollyintegration.cpp

```diff
@@ -282,22 +295,75 @@ QStringList PollyIntegration::get_project_upload_url_commands(QString url_with_w
         QString copy_url_with_wildcard = url_with_wildcard;
         QString url_map_json = copy_url_with_wildcard.replace("*", new_filename) ;
         QString upload_command = "upload_project_data";
-        QList<QByteArray> patch_id_result_and_error = runQtProcess(upload_command, QStringList() << url_map_json << filename);
-        patchId.append(patch_id_result_and_error.at(0));
+        QList<QByteArray> resultAndError = runQtProcess(upload_command, QStringList() << url_map_json << filename);
+        
+        if (_hasError(resultAndError))
+            return patchId;
+
+        patchId.append(resultAndError.at(0));
     }
     return patchId;
 }
 
-bool PollyIntegration::activeInternet()
+bool PollyIntegration::_hasError(QList<QByteArray> resultAndError)
 {
-    QString command = QString("check_internet_connection");
-    QList<QByteArray> results = runQtProcess(command, QStringList());
-    if(!results.isEmpty()) {
-        results = results.at(0).split('\n');
-        QString status = results[1];
-        if (status == "Connected") {
+    QString supportMessage = "Contact tech support at elmaven@elucidata.io if the problem persists";
+    if (resultAndError.size() > 1) {
+        //if there is standard error look for error message
+        QByteArray errorResponse = resultAndError.at(1);
+        errorResponse.replace(QByteArray("\n"), QByteArray(""));
+
+        QJsonDocument doc(QJsonDocument::fromJson(errorResponse));
+        QJsonObject jsonObj = doc.object();
+        QString message;
+        QString type;
+        for (auto key : jsonObj.keys()) {
+            if (key == "message") {
+                message = jsonObj[key].toString();
+            }
+            if (key == "type") {
+                type = jsonObj[key].toString();
+            }
+        }
+
+        if (!message.isEmpty() || !type.isEmpty()) {
+            QString errorMessage = message + "\n" +
+                                   "Type: " + type + "\n" +
```

Comment:
Can we show "Type" only when the `type` string actually has some content? It appears as if we missed some information by showing empty _Type_ row.

<img width="1013" alt="Screenshot 2019-05-15 at 5 34 45 PM" src="https://user-images.githubusercontent.com/10397162/57774288-c267cd00-7737-11e9-8d8e-b35fdf6c73be.png">